Source/qol/stash.cpp

```diff
@@ -698,8 +678,9 @@ bool AutoPlaceItemInStash(Player &player, const Item &item, bool persistItem)
 			if (persistItem) {
 				Stash.stashList.push_back(item);
 				uint16_t stashIndex = static_cast<uint16_t>(Stash.stashList.size() - 1);
+				// move starting position to bottom-left cell
 				Stash.stashList[stashIndex].position = stashPosition + Displacement { 0, itemSize.height - 1 };
-				AddItemToStashGrid(pageIndex, stashPosition, stashIndex, itemSize);
+				UpdateStashGrid(pageIndex, Stash.stashList[stashIndex].position, itemSize, stashIndex + 1);

```

Comment:
Ordinarily I'd be against requiring callers adjust the item index to match the specific implementation (i.e. doing the +1 here instead of in the method) however given `UpdateStashGrid` is effectively an internal function it seems fine as long as you mark it private.

Reply:
(Apologies for the delay, work got messy)
I actually think the stash should be reworked a bit further as so many of the functions littered throughout the stash header/cpp files expected publicly accessible everything (which feels like it explains the struct vs class choice). I have no issue making these additional changes so it's a bit closer to a "proper" public/private usage model, but I wanted to make sure you/others agree that it is the right decision. 

A stopgap measure would be to make the function private, then add public "helper" functions that do the last second conversion (+1 or -1) before calling the private UpdateStashGrid function.